src/components/Carousel/index.tsx

```diff
@@ -0,0 +1,299 @@
+import Container from '@components-layout/Container';
+import { CAROUSEL_SLIDE_STYLE } from '@constants/carouselSlide';
+import styled from '@emotion/styled';
+import { pixelToRem } from '@utils/pixelToRem';
+import { DefaultPropsWithChildren } from '@utils/types/DefaultPropsWithChildren';
+import { debounce, throttle } from 'lodash';
+import {
+  useState,
+  useRef,
+  useEffect,
+  useContext,
+  createContext,
+  Children,
+} from 'react';
+import { MdArrowBackIosNew, MdArrowForwardIos } from 'react-icons/md';
+
+import NavigationButton from './NavigationButton';
+
+interface CarouselProps extends DefaultPropsWithChildren<HTMLDivElement> {
+  line: number;
+  width?: number | undefined;
+  height?: number | undefined;
+}
+
+interface CarouselContextInterface {
+  line: number;
+  cardWidth: number;
+  cardHeight: number;
+  gap: number;
+  translateX: number;
+}
+
+const CarouselContext = createContext<CarouselContextInterface | null>(null);
+
+const isSmallViewPort = (cardWidth: number) => {
+  return window.innerWidth / cardWidth <= 1.2;
+};
+
+const getCardSize = ({
+  line,
+  height,
+  width,
+}: Omit<CarouselProps, 'children'>) => {
+  const cardSize = CAROUSEL_SLIDE_STYLE[line === 1 ? 'inline' : 'multiline'];
+  const newCardSize = { ...cardSize };
+  height && (newCardSize.cardHeight = height);
+  width && (newCardSize.cardWidth = width);
+  if (isSmallViewPort(newCardSize.cardWidth)) {
+    newCardSize.gap = window.innerWidth - newCardSize.cardWidth;
+    return { ...newCardSize, translateX: newCardSize.gap / 2 };
+  }
+  return {
+    ...newCardSize,
+    translateX: newCardSize.gap + Math.floor(newCardSize.cardWidth / 5),
+  };
+};
+
+const useCarouselContext = () => {
+  const context = useContext(CarouselContext);
+  if (context === null) {
+    throw new Error('useTabsContext should be used within Tabs');
+  }
+  return context;
+};
+
+const Carousel = ({ line = 1, children, width, height }: CarouselProps) => {
+  const [currentPage, setCurrentPage] = useState(0);
+  const [sliderWidth, setSliderWidth] = useState(0);
+  const scrollRef = useRef<HTMLDivElement>(null);
+  const cardsRef = useRef<HTMLDivElement>(null);
+  const totalChildren = Children.count(children);
+
+  const { cardWidth, gap, cardHeight, translateX } = getCardSize({
+    line,
+    height,
+    width: width ? width : cardsRef.current?.children[currentPage].clientWidth,
+  });
+  const totalSlide = Math.ceil(totalChildren / line);
+
+  const contextValues = {
+    line,
+    cardWidth,
+    cardHeight,
+    gap,
+    translateX,
+  };
+
+  const scrollToPage = (current: number) => {
+    throttle(() => {
+      current < 0 ? (current = 0) : current;
+      current >= totalSlide ? (current = totalSlide - 1) : current;
+      setCurrentPage(current);
+    }, 200);
+    setTimeout(() => {
+      cardsRef.current?.children[current * line].scrollIntoView({
+        inline: 'start',
+        behavior: 'smooth',
+      });
+    });
+  };
```

Comment:
죄송해요 선민님.. 놓쳤어요 😭
지금 throttle 안의 코드가 작동하지 않아요. 실제로 `currentPage`가 이동하는 이유는 `scrollEventHandler` 때문이에요. 너무 죄송해서 suggestion 가지고 왔어요..  
시간을 500으로 늘린 이유는 smooth scroll에 걸리는 시간이 200ms가 넘어가서 throttle하는 의미가 없어 보여서 넉넉하게 줬어요. cardWidth 값에 따라 스크롤에 걸리는 시간이 조금씩 다른데 적정선을 못찾겠어요.

```suggestion
  const scrollToPage = throttle((current: number) => {
    setTimeout(() => {
      cardsRef.current?.children[current * line].scrollIntoView({
        inline: 'start',
        behavior: 'smooth',
      });
    });
  }, 500);
```

Reply:
확인했습니당